services/mirror/mirror.go

```diff
@@ -293,29 +296,37 @@ func Password(m *models.Mirror) string {
 }
 
 // Update checks and updates mirror repositories.
-func Update() {
+func Update(ctx context.Context) {
 	log.Trace("Doing: Update")
-
 	if err := models.MirrorsIterate(func(idx int, bean interface{}) error {
 		m := bean.(*models.Mirror)
 		if m.Repo == nil {
 			log.Error("Disconnected mirror repository found: %d", m.ID)
 			return nil
 		}
-
-		mirrorQueue.Add(m.RepoID)
-		return nil
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("Aborted due to shutdown")
+		default:
+			mirrorQueue.Add(m.RepoID)
```

Comment:
When `mirrorQueue` is at max capacity, this will block. But `ctx.Done()` can't be checked while blocked. In that scenario, in the event of shutdown this will only break when/if the chan is closed, I believe? I'm not sure this is the right logic.

Reply:
OK I've handled this.